### ISSUE TITLE
Output tensor values differ between using ShardedFlashLlama or using FlashLlama 

### ISSUE THREAD

**Issue by frankxyy** [huggingface/text-generation-inference#324]
I feed the same input to ShardedFlashLlama and FlashLlama of parameter size 13b.

For the third layer, the output tensor for ShardedFlashLlama is :
![image](https://github.com/huggingface/text-generation-inference/assets/6971044/0e4d17b2-b923-4436-ac01-00aee35f6d32)

The output tensor for FlashLlama is:
![image](https://github.com/huggingface/text-generation-inference/assets/6971044/62a381c3-4eb6-4770-a6fb-4ec94dae0164)

You can see that there is a slight difference between the two.

**Comment by Narsil** [comment]
It's perfectly fine for layers to have slight logits variance, In fact it happens all the time when change PyTorch versions, or using different hardware.

Marking as closed unless you can showcase different results (qualitatively or quantatively) down the line (meaning  on a benchmark).

Thanks a lot for the report !

Cheers !
